### PR TITLE
Uniformize build script with other CUDA packages.

### PR DESCRIPTION
**Pull request by maleadt**
Nonfunctional change, except for a bugfix in `apicall` and not looking in versioned subdirectories of `/usr/local` anymore. The installer targets `/usr/local/cuda`, and if the user wants to use a specific version he/she should set CUDA_HOME; I don't think we should be looking there by default.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -45,56 +45,61 @@ end
...
-        else
-            cudapath = Nullable(ENV[first(cudapath_envs)])
+    cuda_envvars = ["CUDA_PATH", "CUDA_HOME", "CUDA_ROOT"]
+    cuda_envvars_set = filter(var -> haskey(ENV, var), cuda_envvars)
+    if length(cuda_envvars_set) > 0
+        cuda_paths = unique(map(var->ENV[var], cuda_envvars_set))
+        if length(unique(cuda_paths)) > 1
+            warn("Multiple, non-equivalent CUDA path variables found in environment. ",
+                 "Arbitrarily selecting CUDA at $(first(cuda_paths)). ",
+                 "To ensure a consistent path, ensure only one CUDA path environment variable is set.")
```
Comment (vchuravy) [inline deps/build.jl]: Maybe mention which cuda environment vars are set? It is won't be clear for people what "one CUDA path" environment is.

File: deps/build.jl [hunk 2]
```diff
@@ -45,56 +45,61 @@ end
...
     if isempty(libcudart)
+        # take a guess at where CUDA is installed
         if is_windows()
-            # location of cudart64_xx.dll or cudart32_xx.dll have to be in PATH env var
-            # ex: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v6.5\bin (by default, it is done by CUDA toolkit installer)
+            # location of cudart64_xx.dll or cudart32_xx.dll has to be in PATH env var
+            # eg. C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v6.5\bin
+            # (by default, it is set by CUDA toolkit installer)
             dllnames = Sys.WORD_SIZE == 64 ?
                 ["cudart64_70", "cudart64_65", "cudart64_60", "cudart64_55", "cudart64_50", "cudart64_50_35"] :
```
Comment (vchuravy) [inline deps/build.jl]: Just noticed that we are probably missing 75 and 80 here

Comment (maleadt) [inline deps/build.jl]: Fixed.

File: deps/build.jl (hunk 1 above)
Comment (maleadt) [inline deps/build.jl]: Fixed.

**Pull request merged**